### DISCUSSION TITLE
Have a control trigger PointerEnter while holding down left click

### DISCUSSION THREAD

**Discussion by EncryptedData** [AvaloniaUI/Avalonia#5677] [Q&A] [answered]
I am working on a visual programming like tool in the vein like Unreal Engine's blueprints. I have two ellipses that I want to drag between to connect the nodes, but the PointerEnter doesn't trigger while holding down left click. Is there a way to have this trigger while click and dragging?

![Imgur](https://i.imgur.com/NGIkfnC.gif)

**Comment by kekekeks** [comment, marked as answer]
1) PointerEnter/PointerLeave won't work for touch input
2) We are aware that the current behavior is against the pointer-events spec, but enter/leave won't work for touch input even when we implement it


For now I'd suggest to add
```
var root = (TopLevel)((IVisual)args.Source).GetVisualRoot();
var rootCoordinates = args.GetPosition(root);
var hitTestResult = root.Renderer.HitTest(rootCoordinates, root, null);
```
to your PointerMoved event handler to get the list of elements below the pointer and update their classes accordingly.

**Reply by EncryptedData** [reply]
Hmm that is sad. Anyways, hopefully we can find a solution to that.

For now, after I find the the element in PointerMoved, could I just manually send the PointerEnter/PointerLeave event? The highlight color on the ellipses is a AXAML Style Setter that I would not want to implement in code (if I don't have to).

**Reply by EncryptedData** [reply]
Also, for now with the tool I am making, the people who would be using it are going to be using Mouse/Keyboard on a desktop. Maybe it should be possible to signal to Avalonia (at least for now) that we want this specific behavior.

**Reply by kekekeks** [reply]
>Hmm that is sad

We've implemented implicit pointer capture for all input device types. We might need to revisit that, but that won't happen *at least* till Summer.

>manually send the PointerEnter/PointerLeave event

I'd suggest to manually set/unset a regular class (e. g. `drag-hover`) and use that for styling.
